modules/gg-fleet-statusd/src/entry.c

```diff
@@ -104,14 +114,79 @@ static void connection_status_close_callback(void *ctx, uint32_t handle) {
     // TODO: Add reconnects (on another thread or with timer
 }
 
+static bool is_positive_integer(GglBuffer str) {
+    if (str.len == 0) {
+        return false;
+    }
+
+    for (size_t counter = 0; counter < str.len; counter++) {
+        if (!isdigit(str.data[counter])) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+static uint64_t get_periodic_status_interval(void) {
+    static uint8_t config_mem[32] = { 0 };
+    GglArena alloc = ggl_arena_init(GGL_BUF(config_mem));
+    GglObject interval_obj;
+
+    GglError ret = ggl_gg_config_read(
+        GGL_BUF_LIST(
+            GGL_STR("services"),
+            GGL_STR("aws.greengrass.NucleusLite"),
+            GGL_STR("configuration"),
+            GGL_STR("fleetStatus"),
+            GGL_STR("periodicStatusPublishIntervalSeconds")
+        ),
+        &alloc,
+        &interval_obj
+    );
+
+    GGL_LOGD(
+        "Config read result: %d, interval_obj type: %d",
+        ret,
+        ggl_obj_type(interval_obj)
+    );
+
+    if (ret == GGL_ERR_OK) {
+        uint64_t interval_seconds = 0;
+
+        if (ggl_obj_type(interval_obj) == GGL_TYPE_I64) {
+            interval_seconds = (uint64_t) ggl_obj_into_i64(interval_obj);
+            GGL_LOGD(
+                "Found interval_obj value (int64): %" PRIu64, interval_seconds
+            );
+        } else if (ggl_obj_type(interval_obj) == GGL_TYPE_BUF
+                   && is_positive_integer(ggl_obj_into_buf(interval_obj))) {
+            interval_seconds = (uint64_t
+            ) strtol((char *) ggl_obj_into_buf(interval_obj).data, NULL, 10);
```

Comment:
This does not check `errno` for parsing errors. Also, `long int` may not be 64-bits, so use `strtoimax`, instead.
```suggestion
            GglBuffer interval_buf = ggl_obj_into_buf(interval_obj);
            errno = 0;
            interval_seconds = (uint64_t
            ) strtoimax((char *) interval_buf.data, NULL, 10);
            if((interval_seconds == 0) && (errno != 0)) {
                GGL_LOGE("Failed to parse interval \"%.*s\" as integer (%d)",
                (int) interval_buf.len, interval_buf.data, errno); 
            }
```